src/default-xrp-client.ts

```diff
@@ -311,6 +316,56 @@ class DefaultXRPClient implements XRPClientDecorator {
       return false
     }
   }
+
+  /**
+   * Return the history of payments for the given account.
+   * Note: This method only works for payment type transactions, see: https://xrpl.org/payment.html
+   * Note: This method only returns the history that is contained on the remote node, which may not contain a full history of the network.
+   *
+   * @param address: The address (account) for which to retrive payment history.
+   * @throws: An error if there was a problem communicating with the XRP Ledger.
+   * @return: An array of transactions associated with the account.
+   */
+  public async paymentHistory(address: string): Promise<Array<XRPTransaction>> {
+    const classicAddress = Utils.decodeXAddress(address)
+    if (!classicAddress) {
+      throw new Error(XRPClientErrorMessages.xAddressRequired)
+    }
+
+    const account = this.networkClient.AccountAddress()
+    account.setAddress(address)
+
+    const request = this.networkClient.GetAccountTransactionHistoryRequest()
+    request.setAccount(account)
+
+    const transactionHistory = await this.networkClient.getTransactionHistory(
+      request,
+    )
+
+    const getTransactionResponses = transactionHistory.getTransactionsList()
+
+    // Filter transactions to payments only and convert them to XRPTransactions.
+    // If a payment transaction fails conversion, throw an error.
+    const payments: Array<XRPTransaction> = []
+    // eslint-disable-next-line no-restricted-syntax
+    // eslint-disable-next-line no-underscore-dangle
+    for (const _getTransactionResponse of getTransactionResponses) {
+      const transaction = _getTransactionResponse.getTransaction()
+      switch (transaction?.getTransactionDataCase()) {
+        case Transaction.TransactionDataCase.PAYMENT: {
+          const xrpTransaction = XRPTransaction.from(transaction)
+          if (!xrpTransaction) {
+            throw new Error(XRPClientErrorMessages.paymentConversionFailure)
+          } else {
+            payments.push(xrpTransaction)
+          }
+          break
+        }
+        default:
+      }
```

Comment:
put a comment here like:
```
default:
   // Intentionally do nothing, non-payment type transactions are ignored.
```

That way future programmers know this was intentional  and you didn't get distracted halfway through a PR.